src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -734,6 +737,56 @@ class spell_dru_innervate : public SpellScript
     }
 };
 
+// 117679 - Incarnation (Passive)
+class spell_dru_incarnation : public AuraScript
+{
+    PrepareAuraScript(spell_dru_incarnation);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_DRUID_INCARNATION_TREE_OF_LIFE
+        });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTarget()->HasAura(SPELL_DRUID_INCARNATION_TREE_OF_LIFE))
+            GetTarget()->RemoveAurasDueToSpell(SPELL_DRUID_INCARNATION_TREE_OF_LIFE);
```

Comment:
Just GetTarget()->RemoveAura(SPELL_DRUID_INCARNATION_TREE_OF_LIFE) is enough, no need to check with HasAura